src/Networks/Xds/Blockcore.Networks.Xds/XdsRegTest.cs

```diff
@@ -1,10 +1,201 @@
-﻿namespace Blockcore.Networks.Xds
+﻿using System;
+using System.Collections.Generic;
+using System.Net;
+using Blockcore.Base.Deployments;
+using Blockcore.Consensus;
+using Blockcore.Consensus.Checkpoints;
+using Blockcore.Features.Consensus.Rules.CommonRules;
+using Blockcore.Features.Consensus.Rules.ProvenHeaderRules;
+using Blockcore.Features.Consensus.Rules.UtxosetRules;
+using Blockcore.Features.MemoryPool.Rules;
+using Blockcore.Networks.Xds.Configuration;
+using Blockcore.Networks.Xds.Consensus;
+using Blockcore.Networks.Xds.Deployments;
+using Blockcore.Networks.Xds.Policies;
+using Blockcore.Networks.Xds.Rules;
+using Blockcore.P2P;
+using NBitcoin;
+using NBitcoin.BouncyCastle.Math;
+using NBitcoin.DataEncoders;
+using NBitcoin.Protocol;
+
+namespace Blockcore.Networks.Xds
 {
     public class XdsRegTest : XdsMain
     {
+        const int RegTestMagicNumberOffset = 2;
         public XdsRegTest()
         {
-            // TODO
+            this.Name = nameof(XdsRegTest);
+            this.NetworkType = NetworkType.Regtest;
+            this.CoinTicker = "XDR";
```

Comment:
XDS?

Reply:
XDR is for the regtest coin, XDT is for the testnet coin. That is used in the hrp of the bech32 addresses.

Reply:
ah ok fair enough.

Reply:
are we forced to 3 characters? isn't better XDSR XDST ?

Reply:
I think, for ui design purposes, it's handy to have the same length....

Reply:
I was concerned about collision with other coins because most of the time a coin on the main chain has 3 character so was "polite" to leave free 3char names for other coins
I'm not sure what you mean about ui design for a regtext/testnet coin, if you look at bitcoin on explorers it's reported as BTC most of the time anyway, if you look at strat on chainz it's reported as STRAT-TEST
imo regtest/testnet tickers should be clearly named to show they are not a main coin chain

Reply:
There is no limitations no I would prefer to go even for XDS-TEST (or XDST is also fine)

Reply:
now I changed it to XDST and XDSR....

Reply:
We could also move away from the convention to use the lower-cased CoinTicker as bech32 hrp.

For example:
CoinTickers: rXDS, tXDS, or regXDS, testXDS
bech32: rxds1...., txds1.....